windows/src/developer/TIKE/compile/CompileKeymanWeb.pas

```diff
@@ -506,9 +508,115 @@ function TCompileKeymanWeb.JavaScript_ContextValue(fkp: PFILE_KEY; pwsz: PWideCh
     Result := Result + Format('",%d)', [Cur - StartQuotes]);
 end;
 
+// Used when targeting 10.0+, after the introduction of FullContextMatch/KFCM.
+function TCompileKeymanWeb.JavaScript_FullContextValue(fkp: PFILE_KEY; pwsz: PWideChar): string;
+
+  function IsRegExpSpecialChar(ch: WideChar): Boolean;
+  begin
+    Result := Pos('"\^$*+?{}.()|[]/', ch) > 0;
+  end;
+
+var
+  Len: Integer;
+  rec: TSentinelRecord;
+  FullContext, Suffix: string;
+begin
+  Result := '';
+  FullContext := '';
+  Suffix := '';
+  Len := xstrlen(pwsz);
+
+	while pwsz^ <> #0 do
+  begin
+    if FullContext <> ''
```

Comment:
Normal patterns we use for if/then/else:

```pascal
if condition then
  singleLineFunction;

if condition
  then shortFunction
  else shortFunction2;

if condition then
begin
  multiLine1;
  multiLine2;
end
else
begin
  // so on
end;

if condition1 then
  result1
else if condition2 then
  result2
else
  result3;
```

Reply:
The inconsistent one is the second example with `then` and `else` on separate lines. We've used this for a long time and it's actually quite readable, so we'll stick with it, even though it is different to the other patterns (and a somewhat unusual pattern generally speaking).